Sources/MongoSwift/ReadConcern.swift

```diff
@@ -14,65 +14,94 @@ public class ReadConcern: Codable {
         case linearizable
         /// See https://docs.mongodb.com/master/reference/read-concern-snapshot/
         case snapshot
-    }
-
-    /// A pointer to a `mongoc_read_concern_t`.
-    internal var _readConcern: OpaquePointer?
+        case custom(level: String)
+
+        public var rawValue: String {
+            switch self {
+            case .local:
+                return "local"
+            case .available:
+                return "available"
+            case .majority:
+                return "majority"
+            case .linearizable:
+                return "linearizable"
+            case .snapshot:
+                return "snapshot"
+            case .custom(let l):
+                return l
+            }
+        }
 
-    /// The level of this `ReadConcern`, or `nil` if the level is not set.
-    public var level: String? {
-        guard let level = mongoc_read_concern_get_level(self._readConcern) else {
-            return nil
+        public init?(rawValue: String) {
+            switch rawValue {
+            case "local":
+                self = .local
+            case "available":
+                self = .available
+            case "majority":
+                self = .majority
+            case "linearizable":
+                self = .linearizable
+            default:
+                self = .custom(level: rawValue)
+            }
         }
-        return String(cString: level)
     }
 
+    /// The level of this `ReadConcern`, or `nil` if the level is not set.
+    public var level: Level?
+
     /// Indicates whether this `ReadConcern` is the server default.
     public var isDefault: Bool {
-        return mongoc_read_concern_is_default(self._readConcern)
+       if level == nil {
+           return true
+       }
+       return false
     }
 
-    /// Initialize a new `ReadConcern` from a `ReadConcern.Level`.
-    public convenience init(_ level: Level) {
-        self.init(level.rawValue)
+    /// Initializes a new `ReadConcern` by copying a `mongoc_read_concern_t`.
```

Comment:
Since we don't actually do a `copy` of the passed-in `mongoc_read_concern_t` anymore, it might be clearer to say something like "Initializes a new `ReadConcern` with the same level as the provided `mongoc_read_concern_t` or something like that.

Reply:
This initializer can also be internal.